django_saml2_auth/views.py

```diff
@@ -165,11 +167,20 @@ def acs(r):
     target_user = None
     is_new_user = False
 
-    try:
-        target_user = user_model.objects.get(username=user_name)
+    find_user_spec = settings.SAML2_AUTH.get('TRIGGER', {}).get('FIND_USER')
+    if find_user_spec:
+        find_user = import_string(find_user_spec)
+        target_user = find_user(user_identity)
+    else:
+        try:
+          target_user = user_model.objects.filter(username=user_name).first()
+        except user_model.DoesNotExists:
+          target_user = None
```

Comment:
The try-except is not needed here, since `first` returns None rather than throwing an exception.